### DISCUSSION TITLE
List of files?

### DISCUSSION THREAD

**Discussion by vbd** [pkolaczk/fclones#278] [Q&A]
I ran `fclones group .` in a directory where I know that duplicate files exist.
The command results in:
```
fclones.exe:  info: Started grouping
fclones.exe:  info: Scanned 554 file entries
fclones.exe:  info: Found 541 (260.9 GB) files matching selection criteria
fclones.exe:  info: Found 31 (8.5 GB) candidates after grouping by size
fclones.exe:  info: Found 31 (8.5 GB) candidates after grouping by paths
fclones.exe:  info: Found 0 (0 B) candidates after grouping by prefix
fclones.exe:  info: Found 0 (0 B) candidates after grouping by suffix
fclones.exe:  info: Found 0 (0 B) redundant files 

# Report by fclones 0.34.0
# Timestamp: 2024-10-31 11:45:17.869 +0100
# Command: 'd:\apps\tools\fclones.exe' group .
# Base dir: d:\\archive
# Total: 0 B (0 B) in 0 files in 0 groups
# Redundant: 0 B (0 B) in 0 files
# Missing: 0 B (0 B) in 0 files

```
But I don't know how to get a list of files. 

EDIT:
I'm on Windows 11 Pro, running powershell 7.4.6.

**Comment by kapitainsky** [comment]
Your results show that there are no duplicates.

> fclones.exe:  info: Found 31 (8.5 GB) candidates after grouping by size
> fclones.exe:  info: Found 31 (8.5 GB) candidates after grouping by paths

There are potentially 31 dups but 

> fclones.exe:  info: Found 0 (0 B) candidates after grouping by prefix
> fclones.exe:  info: Found 0 (0 B) candidates after grouping by suffix

small block hashing,  either initial data  or at the end of the file rules it out. Step 4 and 5 of the algorithm [used](https://github.com/pkolaczk/fclones?tab=readme-ov-file#the-algorithm) by fclones.

You say that you know that dups exist. Could you double check by generating these files hashes?

**Reply by vbd** [reply]
Thank you for your answer.

A trainee messed up the directory with the measurement data during his work (but anyone could have done this, which is a different topic).

I found the following (and some more files, but only one for now):
```
438256813  2024-09-25 03:28:03  './meas--2023-01-24.zip'
438256813  2024-09-25 03:28:03  './2024-10-31/meas--2023-01-24.zip'
```

The two archives I found are identical. I unarchived them and diffed every single file included. The files contain plain text of measurement data (no unicode etc.).

What I don't understand is this:
```
Get-FileHash '.\meas--2023-01-24.zip'

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          C788B70B8278890172EF0A14BC483530E9CD3BEB7FD7BA1E7C28EBC78293CD6D       T:\in\meas--2023-01-24.zip



Get-FileHash '.\2024-10-31\meas--2023-01-24.zip'

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          BC96934E50582D5C443DA418A1183ED39FA031A30F087B20675A0AF00F6B866E       T:\in\2024-10-31\meas--2023-01-24.zip
```

Get-FileHash is the function from powershell.
Any idea why the Hash differ? It seems that it has nothing to do with fclones.
So sorry in advance for taking up your time.

**Reply by kapitainsky** [reply]
At least we know now that fclones is not malfunctioning:) Thank you for following up on it.

IMO Windows often does weird things I find hard to comprehend but I doubt it changes files on its own whim. As in 99% of cases the issue is most likely between keyboard and chair... Maybe these files contain the same data but were created independently - some time stamp changed, one had relative path and other absolute? which result in different zip etc. I can only speculate as only you have keys for unlocking this puzzle.

**Comment by vbd** [comment]
Pebkac ... I'm writing a small Go script to identify the files and will check them by hand. Again sorry for the effort and expense.
